dist/svg.js

```diff
@@ -1409,23 +1415,14 @@ SVG.FX = SVG.invent({
       this.situation = this.situations.shift()
 
       if(this.situation){
-
-        var fn = function(){
-          if(this.situation instanceof SVG.Situation)
-            this.initAnimations().atStart()
-          else if(this.situation instanceof SVG.Delay)
-            this.dequeue()
-          else
-            this.situation.call(this)
-        }.bind(this)
-
-        // start next animation
-        if(this.situation.delay){
-          setTimeout(function(){fn()}, this.situation.delay)
-        }else{
-          fn()
+        if(this.situation instanceof SVG.Situation) {
+          this.startCurrent()
+        } else {
+          // If it is not a SVG.Situation, then it is a function, we execute it
+          // and then pull the next element from the queue
+          this.situation.call(this)
+          return this.dequeue()
```

Comment:
I see your point, I was seeing this as a way to add a callback after an animation but come to think of it there is already `after` that does this... I will change it back :flushed: